overlord/snapstate/snapstate.go

```diff
@@ -869,6 +870,14 @@ func requiresKmodSetup(snapst *SnapState, compsups []ComponentSetup) bool {
 	return false
 }
 
+type componentTaskSetsForSnapInstall struct {
+	afterValidateTasks        []*state.Task
```

Comment:
I'm confused by this, where do the download tasks go? also it would probably best to <FROM>To<TO> names here like in "postHookToDiscardTasks"